spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/ConfigurationPropertiesBinder.java

```diff
@@ -97,6 +98,14 @@ private Validator getConfigurationPropertiesValidator(ApplicationContext applica
 		if (target.getValue() != null && target.getValue().get() instanceof Validator) {
 			validators.add((Validator) target.getValue().get());
 		}
+		for (Class<? extends Validator> validator : annotation.validators()) {
+			try {
+				validators.add(validator.newInstance());
+			}
+			catch (InstantiationException | IllegalAccessException ex) {
+				throw new RuntimeException(ex);
```

Comment:
Is there a better way to implement this? However, I don't expect it will create a lot of trouble.

Reply:
`BeanUtils.instantiateClass` will create the instance and throw a `BeanInstantiationException` on error.

Reply:
Thanks! I'll refactor it later today.